enterprise/frontend/src/metabase-enterprise/embedding_iframe_sdk/embed.v1.unit.spec.ts

```diff
@@ -0,0 +1,70 @@
+import { MetabaseEmbed } from "./embed.v1";
+
+describe("embed.v1.js script tag for sdk iframe embedding", () => {
+  const defaultSettings = {
+    apiKey: "test-api-key",
+    instanceUrl: "http://localhost:3000",
+    target: "#embed-container",
+  };
+
+  it("throws when target element is not found", () => {
+    expect(() => {
+      new MetabaseEmbed({ ...defaultSettings, target: "#not-existent-target" });
+    }).toThrow('cannot find embed container "#not-existent-target"');
+  });
+
+  it("throws when target element is undefined", () => {
+    expect(() => {
+      // @ts-expect-error -- we are testing for incorrect configuration
+      new MetabaseEmbed({ ...defaultSettings, target: undefined });
+    }).toThrow('cannot find embed container "undefined"');
+  });
+
+  it("throws when api key is not provided", () => {
+    expect(() => {
+      // @ts-expect-error -- we are testing for incorrect configuration
+      new MetabaseEmbed({ ...defaultSettings, apiKey: undefined });
+    }).toThrow("api key and instance url must be provided");
+  });
+
+  it("throws when instance url is not provided", () => {
+    expect(() => {
+      // @ts-expect-error -- we are testing for incorrect configuration
+      new MetabaseEmbed({ ...defaultSettings, instanceUrl: undefined });
+    }).toThrow("api key and instance url must be provided");
+  });
+
+  it("throws when both question id and dashboard id are provided", () => {
+    expect(() => {
+      new MetabaseEmbed({
+        ...defaultSettings,
+        questionId: 10,
+        dashboardId: 10,
+      });
+    }).toThrow("can't use both dashboardId and questionId at the same time");
+  });
+
+  it("throws when question id is provided in the exploration template", () => {
+    expect(() => {
+      new MetabaseEmbed({
+        ...defaultSettings,
+        template: "exploration",
+        questionId: 10,
+      });
+    }).toThrow(
+      "the exploration template can't be used with dashboardId or questionId",
+    );
+  });
+
+  it("throws when dashboard id is provided in the exploration template", () => {
```

Comment:
should there be any validation for the `curation` template?

Reply:
we didn't implement the `curation` template yet as it's a bit more complicated, I'll add that in once we have `curation` in the next milestone

Reply:
ok cool!